packages/rpc-provider/src/ws/Provider.ts

```diff
@@ -84,17 +86,22 @@ export default class WsProvider implements WSProviderInterface {
   #websocket: WebSocket | null;
 
   /**
-   * @param {string}  endpoint    The endpoint url. Usually `ws://ip:9944` or `wss://ip:9944`
+   * @param {string | string[]}  endpoint    The endpoint url. Usually `ws://ip:9944` or `wss://ip:9944`, may provide an array of endpoint strings.
    * @param {boolean} autoConnect Whether to connect automatically or not.
    */
-  constructor (endpoint: string = defaults.WS_URL, autoConnectMs: number | false = 1000) {
-    assert(/^(wss|ws):\/\//.test(endpoint), `Endpoint should start with 'ws://', received '${endpoint}'`);
+  constructor (endpoint: string | string[] = defaults.WS_URL, autoConnectMs: number | false = 1000) {
+    endpoint = (typeof endpoint === 'string') ? [endpoint] : endpoint;
+    assert(endpoint.length !== 0, 'WsProvider requires at least one Endpoint');
+    endpoint.forEach((endpoint) => {
+      assert(/^(wss|ws):\/\//.test(endpoint), `Endpoint should start with 'ws://', received '${endpoint}'`);
+    });
```

Comment:
```suggestion
    const endpoints = isString(endpoint)
      ? [endpoint]
      : endpoint;
    
    assert(endpoints.length !== 0, 'WsProvider requires at least one Endpoint');
    
    endpoints.forEach((endpoint) => {
      assert(/^(wss|ws):\/\//.test(endpoint), `Endpoint should start with 'ws://', received '${endpoint}'`);
    });
```

(`isString` from `@polkadot/util`)

Reply:
Alternatively use `Array.isArray` and just swap the `[endpoint]` and `endpoint`.

Reply:
I went with `Array.isArray`, I felt like that read better than isString()